docs/manuals/scripts/generate-supported-versions-timeline.py

```
@@ -0,0 +1,486 @@
+        if supported_segment:
+            supported_x = x_position(supported_segment[0], window_start, window_end)
+            supported_width = (
+                x_position(supported_segment[1], window_start, window_end) - supported_x
+            )
+            parts.append(
+                f'      <rect class="{supported_class}" x="{supported_x}" '
+                f'y="{row_y}" width="{supported_width}" '
+                f'height="{BAR_HEIGHT}" rx="4"/>\n'
+            )
+            text_class = "small-label" if release.projected else "label"
+            text_value = "supported" if release.projected else "supported"
```

Ruff (0.15.15)

[warning] Useless `if`-`else` condition

(RUF034)